test/Test/Toml/Parsing/Unit/Toml.hs

```diff
@@ -8,109 +9,120 @@ import Data.Text (Text)
+            parseToml "[[array]]" $ mkToml $ tableArray "array" (empty :| [])
         it "can parse an array of key/values" $ do
-            let array = tomlFromArray [(makeKey ["array"], NE.fromList [strT, intT])]
+            let array = mkToml $
+                        tableArray "array" $
+                            "key1" =: Text "some string" :|
+                            ["key2" =: Integer 123]
+
             parseToml "[[array]]\n key1 = \"some string\"\n \
-                      \[[array]]\n key2 = 123" array
+                       \[[array]]\n key2 = 123" array
         it "can parse an array of tables" $ do
-            let table1 = tomlFromTable [(makeKey ["table1"], strT)]
-                table2 = tomlFromTable [(makeKey ["table2"], intT)]
-                array  = tomlFromArray [(makeKey ["array"], NE.fromList [table1, table2])]
+            let table1 = table "table1" ("key1" =: Text "some string")
+                table2 = table "table2" ("key2" =: Integer 123)
+                array = mkToml $ tableArray "array" $ table1 :| [table2]
+
             parseToml "[[array]]\n[array.table1] \n key1 = \"some string\"\n \
-                      \[[array]]\n[array.table2] \n key2 = 123" array
+                       \[[array]]\n[array.table2] \n key2 = 123" array
         it "can parse an array of array" $ do
-            let arr = tomlFromArray [(makeKey ["subarray"], NE.fromList [strT, intT])]
-                array = tomlFromArray [(makeKey ["array"], arr :| [])]
+            let arr = tableArray "subarray" ("key1" =: Text "some string" :| ["key2" =: Integer 123])
+                array = mkToml $ tableArray "array" (arr :| [])
+
             parseToml "[[array]] \n [[array.subarray]] \nkey1 = \"some string\"\n \
-                      \[[array.subarray]] \nkey2 = 123" array
-        it "can parse an array of arrays" $ do
-            let arr1 = (makeKey ["table-1"], strT :| [])
-                arr2 = (makeKey ["table-2"], intT :| [])
-                array = tomlFromArray [(makeKey ["array"], tomlFromArray [arr1, arr2] :| [])]
-            parseToml "[[array]]\n [[array.table-1]] \nkey1 = \"some string\"\n \
-                                  \[[array.table-2]] \nkey2 = 123" array
+                       \[[array.subarray]] \nkey2 = 123" array
+        -- TODO this does not work currently, Edsl does not seem to be able to generate arrays of tables with more than one different property
+        --it "can parse an array of arrays" $ do
+            --let
+                --arr1 = tableArray "table-1" ("key1" =: Text "some string" :| [])
+                --arr2 = tableArray "table-2" ("key2" =: Integer 123 :| [])
+                --array = mkToml $ tableArray "array" $ arr1 :| [arr2]
+
+            --parseToml "[[array]]\n [[array.table-1]] \nkey1 = \"some string\"\n \
+                                    -- \[[array.table-2]] \nkey2 = 123" array
```

Comment:
@Nimor111 Could you please clarify this issue a bit more? It's not immediately clear what's the problem here... Could the following examples of using `tableArray` help?

https://github.com/kowainik/tomland/blob/5ad363a001b47fd30e81971457312233c7b8612c/test/Test/Toml/Printer/Golden.hs#L29-L34

I understand that the documentation always can be improved (even if we already have a lot of it!). Maybe I'm just missing something here :thinking:

Reply:
Well, in these examples the structure of the toml is the following: 

```toml
[[deeper]]
ping = "pong"

[[deeper]]

[[deeper]]
green = true

[[deeper]]
[deeper.blue]
red = 255
```

And the expected string here is 
```toml
[[array]]
[[array.table-1]]
key1 = "some string"

[[array.table-2]]
key2 = 123
```

In both of the examples from the golden tests there is only a single thing as a property of the array.
What I got to work was this:

```toml
[[array]]
[[array.table-1]]
key1 = "some string"

[[array]]
[[array.table-2]]
key2 = 123
```

Reply:
I tried in GHCi and the following works for me:

```haskell
mkToml $
    tableArray "array" $
        tableArray "table-1" ("key1" =: "some string" :| [])
        :| [ Edsl.tableArray "table-2" ("key2" =: 123 :| []) ]
```

Reply:
This generates this: 

```toml
[[array]]
[[array.table-1]]
key1 = "some string"

[[array]]
[[array.table-2]]
key2 = 123
```
And the expected string is 

```toml
[[array]]
[[array.table-1]]
key1 = "some string"

[[array.table-2]]
key2 = 123
```
Which sadly is not exactly the same. :/

Reply:
@Nimor111 Ah, I see the difference now. In that case the following should work:

```haskell
mkToml $ tableArray "array" $
    (tableArray "table-1" ("key1" =: "some string" :| []) >> tableArray "table-2" ("key2" =: 123 :| []))
    :| []
```

Reply:
Works great, thanks!